numberformat_proposed.html

```diff
@@ -463,6 +482,84 @@ <h1>UnwrapNumberFormat( _nf_ )</h1>
       </emu-alg>
       <emu-note>See <emu-xref href="#legacy-constructor"></emu-xref> for the motivation of the normative optional text.</emu-note>
     </emu-clause>
+
+    <emu-clause id="sec-setnumberformatunitoptions" aoid="SetNumberFormatUnitOptions">
+      <h1>SetNumberFormatUnitOptions ( _intlObj_, _options_ )</h1>
+      <p>
+        The abstract operation SetNumberFormatUnitOptions resolves the user-specified options in the units realm onto the unit options object.
+      </p>
+      <emu-alg>
+        1. Assert: Type(_intlObj_) is Object.
+        1. Assert: Type(_options_) is Object.
+        1. Let _style_ be ? GetOption(_options_, `"style"`, `"string"`, &laquo; `"decimal"`, `"percent"`, `"currency"`, `"unit"` &raquo;, `"decimal"`).
+        1. Set _intlObj_.[[Style]] to _style_.
+        1. Let _currency_ be ? GetOption(_options_, `"currency"`, `"string"`, *undefined*, *undefined*).
+        1. Let _currencyDisplay_ be ? GetOption(_options_, `"currencyDisplay"`, `"string"`, &laquo; `"code"`, `"symbol"`, `"narrowSymbol"`, `"name"` &raquo;, `"symbol"`).
+        1. Let _unit_ be ? GetOption(_options_, `"unit"`, `"string"`, *undefined*, *undefined*).
+        1. Let _unitDisplay_ be ? GetOption(_options_, `"currencyDisplay"`, `"string"`, &laquo; `"short"`, `"narrow"`, `"long"` &raquo;, `"short"`).
+        1. Let _currencySignDisplay_ be ? GetOption(_options_, `"currencySignDisplay"`, `"string"`, &laquo; `"standard"`, `"accounting"` &raquo;, `"standard"`).
+        1. If _style_ is `"decimal"`, then
+          1. Set _intlObj_.[[UnitType]] to `"none"`.
+        1. If _style_ is `"currency"`, then
+          1. If _currency_ is *undefined*, throw a *TypeError* exception.
+          1. If the result of IsWellFormedCurrencyCode(_currency_) is *false*, throw a *RangeError* exception.
+          1. Let _currency_ be the result of converting _currency_ to upper case as specified in <emu-xref href="#sec-case-sensitivity-and-case-mapping"></emu-xref>.
+          1. Set _intlObj_.[[UnitType]] to `"currency"`.
+          1. Set _intlObj_.[[Currency]] to _currency_.
+          1. Set _intlObj_.[[CurrencyDisplay]] to _currencyDisplay_.
+          1. Set _intlObj_.[[CurrencySignDisplay]] to _currencySignDisplay_.
+        1. If _style_ is `"percent"`, then
+          1. Let _unit_ be `"concentr-percent"`.
+        1. If _style_ is `"unit"` or _style_ is `"percent"`, then
+          1. If _unit_ is *undefined*, throw a *TypeError* exception.
+          1. If the result of IsWellFormedUnitIdentifier(_unit_) is *false*, throw a *RangeError* exception.
+          1. Set _intlObj_.[[UnitType]] to `"measure-unit"`.
```

Comment:
I had considered that.  The reason I used "measure-unit" internally was because I also created an internal field "UnitType", which has values "currency", "measure-unit", or "none", which are each a specific class of units.  This internal vocabulary is influenced by ICU.